### PR TITLE
Added Funding Request Role object, fields, layouts

### LINKED ISSUE #21: Contact Roles
Consider creating a "Contact Roles" child object for Funding Requests

### Use Cases

> The org I am working with wants to automate the generation of their grant letters directly from Salesforce and as part of that, could have anywhere between 2 to 5+ contacts that would need to sign the document. So a contact role type of object would be best.

> In my use case we have many more people than just the one 'Organization Contact' that we need to track - for example, we need to know who the person is who can sign the grant letter, but other people often need to be cc'ed on the grant letter too. Also, just curious if anyone has done a DocuSign integration yet for their contracts for their awarded outbound funds

### Requirements
**Object Name**
- Funding Request Roles

**Fields**
- Contact (Lookup)
- Role (picklist)

### PR DESCRIPTION
**Pull request by lzmls**
# Changes
*https://docs.google.com/spreadsheets/d/1FYGOP1I4HTxnE5fzKTRhA_93uA5zYaEGirgpJZjdVAo/edit#gid=596430561 
*Created Funding Request Object and related fields (see above)
*Created page layouts
*Updated Funding Request Page layout to add related list

# Issues Closed
#21

### CODE REVIEW

**Comment by lzmls** [comment]
@Nickers3 @jtan-trac this is the first thing to come from the Sprint today! Can you review and test? Thanks! Let me or Shari know if you have questions.

**Comment by lzmls** [comment]
I noticed we did not update the lightning page for Funding Requests to include the related object created in this branch (Funding Request Role). It was added to the page layout, but not the lightning page.

File: src/objects/Funding_Request_Role__c.object [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+                    <fullName>Former</fullName>
+                    <default>false</default>
+                    <label>Former</label>
+                </value>
+            </valueSetDefinition>
+        </valueSet>
+    </fields>
+    <label>Funding Request Role</label>
+    <nameField>
+        <displayFormat>FR-{0000}</displayFormat>
```
Comment (umeditor) [inline src/objects/Funding_Request_Role__c.object]: I'd suggest adding some more zeros; I imagine having lots of roles. Some like FR-{000000} would be more future-proof.

**Comment by Nickers3** [comment]
@lzmls @reillyellis finally got around at looking into the build.  Some questions that came to mind are:

1) Do we want a tab for the object?
2) For the search/lookup dialogs, should contact be included?
3) On the Funding Request Lightning Page, should this be a tab similar to Requirements and Disbursements?  The screenshot showed it above Activities.

**Comment by Nickers3** [comment]
@lzmls @reillyellis one more idea - would it make sense to add a Quick Action to the Funding Request to add Roles?  With the related list front and centered, it may not be needed.  Thought I'd throw the idea out there.

**Comment by alexasgotsaas** [comment]
I tested the Funding Request Role feature. I did not encounter any issues with adding Funding Request Roles. A few improvement suggestions:
1. Currently you can mark primary for Funding Request roles of the same type (ie you can have two Grant Managers marked primary) Consider removing Primary checkbox or adding functionality to limit the number of Primary allowed. Additionally if the Primary is meant to indicate one Primary Funding Request Contact (instead of one per type), consider adding clarification in the field Label "Primary Funding Request Contact"
2. Link Funding Request Roles to Requirements and Disbursements. Requirements has a Contact Lookup. Contacts linked here should also be Funding Request Roles, create/change functionality to eliminate duplicate entry and potential for data discrepancy. Financial Managers should be connected to Disbursements, create functionality to link specific Contacts/Roles to Disbursements
3. For the out of the box picklist values for Role, how were these determined? Since terminology is important, you may want to get several foundations' input. I've more commonly seen Applicant, Financial Contact, Reporting Contact, Executive Director, Board Member

**Comment by Nickers3** [comment]
> I tested the Funding Request Role feature. I did not encounter any issues with adding Funding Request Roles. A few improvement suggestions:
> 
> 1. Currently you can mark primary for Funding Request roles of the same type (ie you can have two Grant Managers marked primary) Consider removing Primary checkbox or adding functionality to limit the number of Primary allowed. Additionally if the Primary is meant to indicate one Primary Funding Request Contact (instead of one per type), consider adding clarification in the field Label "Primary Funding Request Contact"
> 2. Link Funding Request Roles to Requirements and Disbursements. Requirements has a Contact Lookup. Contacts linked here should also be Funding Request Roles, create/change functionality to eliminate duplicate entry and potential for data discrepancy. Financial Managers should be connected to Disbursements, create functionality to link specific Contacts/Roles to Disbursements
> 3. For the out of the box picklist values for Role, how were these determined? Since terminology is important, you may want to get several foundations' input. I've more commonly seen Applicant, Financial Contact, Reporting Contact, Executive Director, Board Member

Thanks for the thoughts!

**Comment by Nickers3** [comment]
Recapping some of the items from the call today:

1) We will take out Primary as a field in this build.  There is a lot of functionality that could be added such as limited the number of primaries per request that open up the need for code so we will add this as an enhancement for a later build.  Taking it out would eliminate confusion of the use case(s) and expectation of automation included.

2) The linking of Funding Request Role to Requirements and Disbursements will be added as an enhancement.  Taking into consideration, the need for someone to use a Community, it will be best to keep the lookups to be for a contact.  There could be automation that adds a Funding Request Role if a contact is linked so marking as an enhancement would be best.

3) For the roles, the initial list was from a brainstorm at the Philadelphia Sprint.  While more roles could be added, it is best to have fewer than more.  With each use case being different, having less means less work for admins to deactivate to fit their use case.

**Pull request merged**